### PR TITLE
Add workaround for race condition between agetty and logind

### PR DESCRIPTION
**Pull request by mloviska**
As described in https://github.com/systemd/systemd/issues/23660 or [Bug 1185098 - [Build 15.82]
openQA test fails in snapper_undochange - logind fails after return back from rescue](https://bugzilla.suse.com/show_bug.cgi?id=1185098)
we need to add workaround that ensures `tty2` will be available for login.
Issue occurs only in systems running sle15sp2 or below.

VRs:
    - [workaround case](http://kepler.suse.cz/tests/17406#step/snapper_undochange/55)
    - [expected](http://kepler.suse.cz/tests/17405#step/snapper_undochange/42)

### CODE REVIEW

File: lib/btrfs_test.pm [hunk 1]
```diff
@@ -72,6 +74,17 @@ sub snapper_nodbus_restore {
     die 'DBus service should be inactive, but it is active' if ($ret == 0);
     script_run('systemctl default', timeout => 600, die_on_timeout => 0);
     my $tty = get_root_console_tty;
+
+    if (is_sle('<15-SP3') && !defined(my $match = check_screen("tty$tty-selected", 120))) {
+        record_soft_failure("bsc#1185098 - [Build 15.82] openQA test fails in snapper_undochange - logind fails after return back from rescue");
```
Comment (jlausuch) [inline lib/btrfs_test.pm:79]: nitpicking, no need to provide full title of the bug, but feel free to reject :)
```suggestion
        record_soft_failure("bsc#1185098 - logind fails after return back from rescue");
```

File: lib/btrfs_test.pm [hunk 2]
```diff
@@ -72,6 +74,17 @@ sub snapper_nodbus_restore {
...
+        select_console('log-console');
+        if (script_run('systemctl is-active getty@tty2.service') == 3) {
```
Comment (jlausuch) [inline lib/btrfs_test.pm:81]: We should have the best practice of `script_run` together with `die_on_timeout` flag.
https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1010

```suggestion
        if (script_run('systemctl is-active getty@tty2.service', die_on_timeout => 1) == 3) {
```

File: lib/btrfs_test.pm (hunk 1 above)
Comment (mloviska) [reply lib/btrfs_test.pm]: Nah, I that was just my lazy copy&paste. Updated :)

File: lib/btrfs_test.pm (hunk 2 above)
Comment (mloviska) [reply lib/btrfs_test.pm]: Nice catch, I always forget about this option. Thanks!

Comment (mloviska) [reply lib/btrfs_test.pm]: http://kepler.suse.cz/tests/17407
http://kepler.suse.cz/tests/17408

File: lib/btrfs_test.pm [hunk 3]
```diff
@@ -72,6 +74,17 @@ sub snapper_nodbus_restore {
     die 'DBus service should be inactive, but it is active' if ($ret == 0);
     script_run('systemctl default', timeout => 600, die_on_timeout => 0);
     my $tty = get_root_console_tty;
+
+    if (is_sle('<15-SP3') && !defined(my $match = check_screen("tty$tty-selected", 120))) {
+        record_soft_failure("bsc#1185098 - logind fails after return back from rescue");
```
Comment (jlausuch) [inline lib/btrfs_test.pm:79]: Also, Oli had a valid point about WON'T FIX bugs, this seems to be one of them. By doing soft-failure, our jobs will be always soft-failed and will keep posting comments to the ticket, I wonder if we should just use record_info...

Comment (mloviska) [reply lib/btrfs_test.pm:79]: That is a valid point, we can change that when there will be some response from the upstream. As of now, there are no comments, so maybe they will consider fixing it.

Comment (jlausuch) [reply lib/btrfs_test.pm:79]: Makes sense.

**Review by jlausuch — APPROVED** [review]

**Pull request merged**